### PR TITLE
Use docusaurus

### PR DESCRIPTION
**Pull request by aabedraba**
- Change mkdocs with docusaurus
- Fix all broken links with the help of docusaurus **(more than 50!)**
- Removes roadmap page
- Removes architecture page
- Updated netlify manually
- update github workflow for docusaurus
- add google analytics for docusaurus

Demo website is available here: https://kusk-gateway-docs-2.vercel.app/docs/intro

Example of broken links error message :) 
telegram-cloud-photo-size-4-6032815436418103507-y

### CODE REVIEW

File: docs/Dockerfile [hunk 1]
```diff
@@ -1,28 +0,0 @@
-FROM python:rc-alpine3.12 as builder
```
Comment (kylehodgetts) [inline docs/Dockerfile:1]: Can we keep the dockerfile to spin up a local version of the docs site? It allows us to validate that the docs are correct before pushing to git

Comment (aabedraba) [reply docs/Dockerfile:1]: You can have a local version just by running
`npm run start` 

Do you still think we need the dockerfile for it?

**Review by jasmingacic — APPROVED** [review]

**Review by jasmingacic — APPROVED** [review]

**Review by jasmingacic — APPROVED** [review]

File: .github/workflows/docusaurus.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          # Build output to publish to the `gh-pages` branch:
+          publish_dir: ./build
+          # The following lines assign commit authorship to the official
+          # GH-Actions bot for deploys to `gh-pages` branch:
+          # https://github.com/actions/checkout/issues/13#issuecomment-724415212
+          # The GH actions bot is used by default if you didn't specify the two fields.
+          # You can swap them out with your own user credentials.
+          user_name: github-actions[bot]
+          user_email: 41898282+github-actions[bot]@users.noreply.github.com
```
Comment (kylehodgetts) [inline .github/workflows/docusaurus.yml:43]: new line missing

File: docs/README.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+You can find the docs here: kusk.io/docs
+
+## Edit the docs
+
+If you're editing the docs, follow this workflow:
+
+1- Spin up local development with `npm run start`
+2- Update the docs inside the `/docs` folder (they're good 'ol markdown)
+3- (if there's a new page in the docs) add it to the file `sidebar.js` so the menu on the left is updated
```
Comment (kylehodgetts) [inline docs/README.md:11]:
```suggestion
3- if there's a new page in the docs, add it to the file `sidebar.js` so the menu on the left is updated
```

**Review by kylehodgetts — CHANGES_REQUESTED** [review]

File: build/manager/.dockerignore [hunk 4]
```diff
@@ -0,0 +1 @@
+**/docs
```
Comment (kylehodgetts) [inline build/manager/.dockerignore:1]: newline ;)

**Review by kylehodgetts — APPROVED** [review]

**Pull request merged**